### PR TITLE
Add statistics struct and Support manually inject statistics

### PR DESCRIPTION
**Pull request by EmmyMiao87**
## Proposed changes

This PR mainly developed the data structure used by statistical information
and the function of manually modifying the statistical information.
We use a statistics package alone to store statistical information,
and use the 'statistics manager' as a unified entry for statistical information.
For detailed data structure and explanation, please refer to the comments on the class.

Manually modify statistics include: Manually modify table statistics and column statistics.
The syntax is explained in the issue #6370.

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Code refactor (Modify the code structure, format the code, etc...)
- [ ] Optimization. Including functional usability improvements and performance improvements.
- [ ] Dependency. Such as changes related to third-party components.
- [ ] Other.

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have created an issue on (Fix #ISSUE) and described the bug/feature there in detail
- [x] Compiling and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If these changes need document changes, I have updated the document
- [ ] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at dev@doris.apache.org by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/cup/sql_parser.cup [hunk 1]
```diff
@@ -827,6 +828,20 @@ alter_stmt ::=
     {:
         RESULT = new AlterRoutineLoadStmt(jobLabel, jobProperties, datasourceProperties);
     :}
+    | KW_ALTER KW_TABLE table_name:tbl KW_SET KW_STATS LPAREN key_value_map:map RPAREN
```
Comment (morningman) [inline fe/fe-core/src/main/cup/sql_parser.cup:831]: Could you please add this new statement in #6277 ?
I think they should be under `Database Administration Statements`(same as MySQL)

File: fe/fe-core/src/main/cup/sql_parser.cup [hunk 2]
```diff
@@ -263,7 +263,7 @@ terminal String KW_ADD, KW_ADMIN, KW_AFTER, KW_AGGREGATE, KW_ALIAS, KW_ALL, KW_A
     KW_RIGHT, KW_ROLE, KW_ROLES, KW_ROLLBACK, KW_ROLLUP, KW_ROUTINE, KW_ROW, KW_ROWS,
     KW_S3, KW_SCHEMA, KW_SCHEMAS, KW_SECOND, KW_SELECT, KW_SEMI, KW_SERIALIZABLE, KW_SESSION, KW_SET, KW_SETS, KW_SET_VAR, KW_SHOW, KW_SIGNED,
     KW_SKEW,
-    KW_SMALLINT, KW_SNAPSHOT, KW_SONAME, KW_SPLIT, KW_START, KW_STATUS, KW_STOP, KW_STORAGE, KW_STREAM, KW_STRING, KW_STRUCT,
+    KW_SMALLINT, KW_SNAPSHOT, KW_SONAME, KW_SPLIT, KW_START, KW_STATUS, KW_STATS, KW_STOP, KW_STORAGE, KW_STREAM, KW_STRING, KW_STRUCT,
```
Comment (morningman) [inline fe/fe-core/src/main/cup/sql_parser.cup:266]: Add KW_STATS to `keywords`?

File: fe/fe-core/src/main/java/org/apache/doris/statistics/StatisticsManager.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+    public void alterColumnStatistics(AlterColumnStatsStmt stmt) throws DdlException, AnalysisException {
+        Table table = validateTableName(stmt.getTableName());
+        String columnName = stmt.getColumnName();
+        Column column = table.getColumn(columnName);
+        if (column == null) {
+            ErrorReport.reportDdlException(ErrorCode.ERR_BAD_FIELD_ERROR, columnName);
+        }
+        // match type and column value
+        statistics.updateColumnStats(table.getId(), columnName, column.getType(), stmt.getProperties());
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/statistics/StatisticsManager.java:53]: No need lock?

File: fe/fe-core/src/main/java/org/apache/doris/statistics/OlapTableStats.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+ *     Those @TabletStats are recorded in @idToTabletStats form of MAP.
+ *     This facilitates the optimizer to quickly find the corresponding
+ *     @TabletStats based on the tablet id.
+ *     At the same time, both @rowCount and @dataSize can also be obtained
+ *     from the sum of all @TabletStats.
+ *
+ */
+public class OlapTableStats extends TableStats {
+
+    private Map<Long, TabletStats> idToTabletStats = Maps.newHashMap();
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/statistics/OlapTableStats.java:45]: How about put this to TabletInvertedIndex?

Comment (EmmyMiao87) [reply fe/fe-core/src/main/java/org/apache/doris/statistics/OlapTableStats.java:45]: All classes in statistical information are currently designed to be completely decoupled from other meta-information.

File: fe/fe-core/src/main/cup/sql_parser.cup (hunk 1 above)
Comment (EmmyMiao87) [reply fe/fe-core/src/main/cup/sql_parser.cup:831]: DONE

File: fe/fe-core/src/main/cup/sql_parser.cup (hunk 2 above)
Comment (EmmyMiao87) [reply fe/fe-core/src/main/cup/sql_parser.cup:266]: done

File: fe/fe-core/src/main/java/org/apache/doris/statistics/StatisticsManager.java (hunk 3 above)
Comment (EmmyMiao87) [reply fe/fe-core/src/main/java/org/apache/doris/statistics/StatisticsManager.java:53]: use concurrent map instead

**Review by morningman — DISMISSED** [review]
LGTM

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**